### DISCUSSION TITLE
How to switch between multiple Neovide windows on Mac?

### DISCUSSION THREAD

**Discussion by everdimension** [neovide/neovide#1894] [Q&A] [answered]
I can open several Neovide _windows_ by typing "neovide" in terminal

Usually on mac you can switch between different windows of the same app by pressing `command + ~` (tilde)

But this doesn't work in neovide

Is there a specific way, perhaps using command mode, to switch between open gui windows?

**Comment by corintho** [comment]
I'm currently wondering the same thing.
They behave as totally different running apps, instead of behaving as multiple windows of the same app.
And since all the Apps are named "Neovide" (I understand that is a limitation with OSX) it is quite confusing to find the correct one sometimes

**Comment by fredizzimo** [comment]
@alex35mil actually just released this. https://github.com/alex35mil/NeoHub

It's not ideal, but we are lacking macOS developers, and at least to me it's not even clear how it should work on macOS. Each window is a totally independent process with it's own neovim instance, and on the future we might actually want to support multiple windows per process.

**Reply by corintho** [reply]
I will take a look at it, thanks for the suggestion

**Reply by alex35mil** [reply]
I've been actively using it for quite some time, and it's been working smoothly on my machine.

Also, I'm testing the restart editor feature. Helpful when tweaking a config, or something hanged/messed up along the way.

**Reply by corintho** [reply]
Whoever finds this thread while looking for information on how to switch neovide windos in OSX, I strongly suggest you check out NeoHub.
I've been using it daily, and although it is not perfect (due to limitations with Apple APIs and multiple neovide windows) it drastically improves the workflow. Since it adds a global shortcut to OSX, it is easier now to find my neovide window, than the window for other editors I used in the past.

**Reply by micsama** [reply]
> I will take a look at it, thanks for the suggestion

  I've been a huge fan of @alex35mil's NeoHub—it truly changes the game for Neovide users on macOS. Since the original project hasn't been updated for a while,
  I decided to give it a "Reboot" to keep it compatible with the latest macOS versions and modern development standards.

  I've released [NeoHubR](https://github.com/micsama/NeoHubR) (Reboot), which builds upon the original vision with several enhancements:

   * Modern Core: Completely rewritten using Swift 6 and Swift Concurrency (async/await) for better performance and safety.
   * Improved CLI (`nh`): A more robust CLI tool with better path resolution and IPC reliability.
   * Enhanced UX: A more polished Switcher (Teleport) and project management registry.
   * Architecture Cleanup: Significant refactoring to make the codebase more maintainable.

  I'm still relatively new to Swift and macOS development, but I’ve poured a lot of heart into making this "Reboot" as stable and smooth as possible for daily
  use. 

  If anyone is still looking for a modern instance manager for Neovide, feel free to check it out! Big thanks to @alex35mil again for the amazing original work.
  Feedback and PRs are more than welcome! 🚀

**Comment by ivotron** [comment]
an alternative is to use session/workspace management within neovim. many plugins out there. in my case i'm using https://github.com/folke/persistence.nvim within lazyvim

**Comment by micsama** [comment]
I've been a huge fan of @alex35mil's NeoHub—it truly changes the game for Neovide users on macOS. Since the original project hasn't been updated for a while,
  I decided to give it a "Reboot" to keep it compatible with the latest macOS versions and modern development standards.

  I've released [NeoHubR](https://github.com/micsama/NeoHubR) (Reboot), which builds upon the original vision with several enhancements:

   * Modern Core: Completely rewritten using Swift 6 and Swift Concurrency (async/await) for better performance and safety.
   * Improved CLI (`nh`): A more robust CLI tool with better path resolution and IPC reliability.
   * Enhanced UX: A more polished Switcher (Teleport) and project management registry.
   * Architecture Cleanup: Significant refactoring to make the codebase more maintainable.

  I'm still relatively new to Swift and macOS development, but I’ve poured a lot of heart into making this "Reboot" as stable and smooth as possible for daily
  use. 

  If anyone is still looking for a modern instance manager for Neovide, feel free to check it out! Big thanks to @alex35mil again for the amazing original work.
  Feedback and PRs are more than welcome! 🚀

**Comment by falcucci** [comment, marked as answer]
we have now [multi-window support](https://neovide.dev/configuration.html#macos-multi-window-editors) for macOS on neovide [**nightly**](https://github.com/neovide/neovide/releases/tag/nightly) so you can check it out later :)
